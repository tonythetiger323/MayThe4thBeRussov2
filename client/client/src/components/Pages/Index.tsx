import * as React from 'react'
import Page from '../Layouts/Page'
import Typography from '@material-ui/core/Typography'

class Index extends React.Component {
  render() {
    return (
      <div>
        <Page>
          <Typography variant='h2'>May the 4th be Russo</Typography>
        </Page>
      </div>
    )
  }
}

export default Index
