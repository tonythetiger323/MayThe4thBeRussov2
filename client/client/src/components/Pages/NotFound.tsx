import * as React from 'react'
import Page from '../Layouts/Page'
import Typography from '@material-ui/core/Typography'

const NotFound = (): JSX.Element => {
  return (
    <Page>
      <Typography variant='h1'>Page Not Found</Typography>
    </Page>
  )
}

export default NotFound
