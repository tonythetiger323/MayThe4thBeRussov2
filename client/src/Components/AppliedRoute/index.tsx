// Import dependencies
import * as React from 'react'
import { Route } from 'react-router-dom'

// Creating a route that will apply child props
const AppliedRoute = ({ component: C, props: cProps, ...rest }: any) => {
  return (
    <Route {...rest} render={(props: any) => <C {...props} {...cProps} />} />
  )
}
export default AppliedRoute
