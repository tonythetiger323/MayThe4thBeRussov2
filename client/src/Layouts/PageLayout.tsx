import * as React from 'react'
import Typography from '@material-ui/core/Typography'
import Grid from '@material-ui/core/Grid'
import Navbar from '../Components/Navbar'
import Footer from '../Components/Footer'
import NavbarLink from '../Components/NavbarLink'
import { Parallax } from 'react-parallax'

class PageLayout extends React.Component {
  render() {
    return (
      <div className='app-container'>
        <Parallax
          blur={3}
          bgImage={require('./images/MayThe4thBeRusso_Background Image_sean-o-406693-unsplash.jpg')}
          bgImageAlt={'Hawaiian Beach'}
          strength={100}
        >
        <header>
          <Typography variant='h3' align='center'>
            Megan &amp; Randy
          </Typography>
          <Typography variant='h5' align='center'>May 14th 2019</Typography>
        </header>
        <nav>
          <Grid container alignItems='center' justify='space-around'>
            <Grid item>
              <NavbarLink to='/'>
                <h1>Megan &amp; Randy</h1>
              </NavbarLink>
            </Grid>
            <Grid item>
              <Navbar />
            </Grid>
          </Grid>
        </nav>
        <main>{this.props.children}</main>
        <footer>
          <Footer />
        </footer>
        </Parallax>
      </div>
    )
  }
}
export default PageLayout
