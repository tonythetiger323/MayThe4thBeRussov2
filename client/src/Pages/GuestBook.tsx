import React from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { Image, Transformation } from 'cloudinary-react'
import Paper from '@material-ui/core/Paper'
import MessageDialog from '../Components/MessageDialog'
import { withStyles, Theme } from '@material-ui/core/styles'
import GuestbookMessages from '../Components/GuestBookMessages'
import Responsive from 'react-responsive'

const Mobile = (props: any) => <Responsive {...props} maxWidth={667} />
const Default = (props: any) => <Responsive {...props} minWidth={668} />

const styles = (theme: Theme) => ({
  root: {
    ...theme.mixins.gutters(),
    paddingTop: theme.spacing(2),
    paddingBottom: theme.spacing(2)
  }
})

function GuestBook(props: any) {
  const { classes } = props

  return (
    <div>
      <Grid
        container
        direction="column"
        alignItems="center"
        alignContent="center"
        justify="center"
      >
        <Grid item>
          <Typography variant="h4" align="center">
            Guest Book
          </Typography>
        </Grid>
        <br />
        <Grid item>
          <Default>
            <Image
              cloudName="tonythetiger323"
              publicId="MayThe4thBeRusso/Other Photos/Together/FullSizeRender.jpg"
            >
              <Transformation width="550" crop="scale" />
            </Image>
          </Default>
          <Mobile>
            <Image
              cloudName="tonythetiger323"
              publicId="MayThe4thBeRusso/Other Photos/Together/FullSizeRender.jpg"
            >
              <Transformation width="375" crop="scale" />
            </Image>
          </Mobile>
        </Grid>
        <br />
        <Paper style={{ opacity: 0.6 }} className={classes.root}>
          <Grid
            container
            direction="column"
            alignItems="center"
            alignContent="center"
            justify="center"
          >
            <Grid item>
              <Typography
                style={{ fontFamily: 'Oswald' }}
                variant="h6"
                align="center"
              >
                Help us remember this very special time in our lives by leaving
                us a note! Thank you so much for celebrating with us!
              </Typography>
            </Grid>
            <br />
            <Grid item>
              <MessageDialog />
            </Grid>
          </Grid>
        </Paper>
        <br />
        <Grid item>
          <Typography
            variant="h4"
            style={{ fontFamily: 'Oswald' }}
            align="center"
          >
            Guest Book Messages
          </Typography>
        </Grid>
        <br />
        <Grid item>
          <GuestbookMessages />
        </Grid>
      </Grid>
    </div>
  )
}

export default withStyles(styles)(GuestBook)
