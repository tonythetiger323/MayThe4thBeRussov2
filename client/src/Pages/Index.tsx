import * as React from 'react'
import PhotoStepper from '../Components/PhotoStepper'

class Index extends React.Component {
  render() {
    return (
      <div>
        <PhotoStepper />
      </div>
    )
  }
}

export default Index
