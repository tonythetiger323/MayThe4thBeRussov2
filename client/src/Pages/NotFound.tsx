import * as React from 'react'
import Typography from '@material-ui/core/Typography'

const NotFound = (): JSX.Element => {
  return <Typography variant='h1'>Page Not Found</Typography>
}

export default NotFound
