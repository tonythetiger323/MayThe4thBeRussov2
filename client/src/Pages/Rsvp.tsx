import React from 'react';
import RsvpCard from '../Components/RsvpCard'


class Rsvp extends React.Component{
  render(){
    return(
      <RsvpCard/>
      );
  };
}

export default Rsvp;
