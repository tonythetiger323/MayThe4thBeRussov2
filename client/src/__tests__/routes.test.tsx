import * as React from 'react'
import Routes from '../routes'
import { shallow } from 'enzyme'
import AppliedRoute from '../Components/AppliedRoute'
import PageLayout from '../Layouts/PageLayout'
import Index from '../Pages/Index'
import GuestBook from '../Pages/GuestBook'
import NotFound from '../Pages/NotFound'
import Rsvp from '../Pages/Rsvp'
import { createShallow } from '@material-ui/core/test-utils/index'

describe('routes', () => {
  let routes = shallow(<Routes />)
  const shallowMui = createShallow()

  it('should render correctly', () => {
    expect(routes).toMatchSnapshot()
  })

  describe('AppliedRoute', () => {
    it('should work', () => {
      routes = shallowMui(<AppliedRoute />)
    })
  })

  describe('PageLayout', () => {
    it('should work', () => {
      routes = shallowMui(<PageLayout />)
    })
  })

  describe('Index', () => {
    it('should work', () => {
      routes = shallowMui(<Index />)
    })
  })

  describe('GuestBook', () => {
    it('should work', () => {
      routes = shallowMui(<GuestBook />)
    })
  })

  describe('NotFound', () => {
    it('should work', () => {
      routes = shallowMui(<NotFound />)
    })
  })

  describe('Rsvp', () => {
    it('should work', () => {
      routes = shallowMui(<Rsvp />)
    })
  })
})
