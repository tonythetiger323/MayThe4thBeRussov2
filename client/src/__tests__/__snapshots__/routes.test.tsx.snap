// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should render correctly 1`] = `ShallowWrapper {}`;
